lightning/src/ln/chan_utils.rs

```diff
@@ -550,11 +556,430 @@ pub fn build_htlc_transaction(prev_hash: &Txid, feerate_per_kw: u32, contest_del
 	}
 }
 
+// Per-channel data used to build transactions given per-commitment data (CommitmentTransaction)
+#[derive(Clone)]
+pub(crate) struct ChannelTransactionParameters {
+	// Holder public keys
+	pub(crate) holder_pubkeys: ChannelPublicKeys,
+	// Counter-party public keys
+	pub(crate) counterparty_pubkeys: ChannelPublicKeys,
+	// The contest delay selected by the holder
+	pub(crate) holder_selected_contest_delay: u16,
+	// The contest delay selected by the counterparty
+	pub(crate) counterparty_selected_contest_delay: u16,
+	// The funding outpoint
+	pub(crate) funding_outpoint: OutPoint,
+	// Whether the channel is outbound from the point of view of the holder
+	pub(crate) is_outbound_from_holder: bool,
+}
+
+impl ChannelTransactionParameters {
+	pub(crate) fn to_directed(&self, holder: bool) -> DirectedChannelTransactionParameters {
+		DirectedChannelTransactionParameters {
+			info: self,
+			broadcaster_is_holder: holder
+		}
+	}
+}
+
+impl_writeable!(ChannelTransactionParameters, 0, {
+	holder_pubkeys,
+	counterparty_pubkeys,
+	holder_selected_contest_delay,
+	counterparty_selected_contest_delay,
+	funding_outpoint,
+	is_outbound_from_holder
+});
+
+/// Static channel fields used to build transactions given per-commitment fields (CommitmentTransaction)
+pub struct DirectedChannelTransactionParameters<'a> {
+	/// The holder's channel static info
+	info: &'a ChannelTransactionParameters,
+	/// Whether the holder is the broadcaster
+	broadcaster_is_holder: bool,
+}
+
+impl<'a> DirectedChannelTransactionParameters<'a> {
+	/// Get the pubkeys for the broadcaster and countersignatory
+	pub fn pubkeys(&self) -> (&ChannelPublicKeys, &ChannelPublicKeys) {
+		if self.broadcaster_is_holder {
+			(&self.info.holder_pubkeys, &self.info.counterparty_pubkeys)
+		} else {
+			(&self.info.counterparty_pubkeys, &self.info.holder_pubkeys)
+		}
+	}
+
+	/// Get the contest delay applicable to the transactions.
+	/// Note that the contest delay was selected by the countersignatory.
+	pub fn contest_delay(&self) -> u16 {
+		if self.broadcaster_is_holder { self.info.counterparty_selected_contest_delay } else { self.info.holder_selected_contest_delay }
+	}
+
+	/// Whether the channel is outbound from the broadcaster
```

Comment:
Can you add a comment explaining that channel initiation is used as an ordering for commitment number generation ? Everytime I'm skimming over this code I'm confused why outboundness is leaking at the transaction construction level...